code/game/objects/effects/mines.dm

```diff
@@ -1,25 +1,117 @@
+/obj/item/deployablemine
+	name = "deployable mine"
+	desc = "An unarmed landmine. It can be planted to arm it."
+	icon_state = "uglymine"
+	var/mine_type = /obj/effect/mine
+	var/arming_time = 30
+
+/obj/item/deployablemine/stun
+	desc = "An unarmed stun mine. It can be planted to arm it."
+	mine_type = /obj/effect/mine/stun
+
+/obj/item/deployablemine/smartstun
+	name = "deployable smart mine"
+	desc = "An unarmed smart stun mine. It can be planted to arm it."
+	mine_type = /obj/effect/mine/stun/smart
+
+/obj/item/deployablemine/rapid
+	name = "deployable rapid smart mine"
+	desc = "An unarmed smart stun mine designed to be rapidly placeable."
+	mine_type = /obj/effect/mine/stun/smart/adv
+	arming_time = 10
+	w_class = WEIGHT_CLASS_SMALL
+
+/obj/item/deployablemine/heavy
+	name = "deployable sledgehammer smart mine"
+	desc = "An unarmed smart heavy stun mine designed to be hard to disarm."
+	mine_type = /obj/effect/mine/stun/smart/heavy
+	arming_time = 50
+
+/obj/item/deployablemine/shrapnel
+	name = "deployable Sharpnel  mine"
+	desc = "An unarmed sharpnel  mine designed to spray lead everywhere and  be hard to disarm."
+	mine_type = /obj/effect/mine/shrapnel
+	arming_time = 50
+
+/obj/item/deployablemine/explosive
+	mine_type = /obj/effect/mine/explosive
+
+
+/obj/item/deployablemine/gas
+	name = "oxygen gas mine"
+	desc = "An unarmed mine that releases oxygen into the air when triggered. Pretty pointless huh."
+	mine_type = /obj/effect/mine/gas
+
+/obj/item/deployablemine/plasma
+	name = "incendiary mine"
+	desc = "An unarmed mine that releases plasma into the air when triggered, then ignites it."
+	mine_type = /obj/effect/mine/gas/plasma
+
+/obj/item/deployablemine/sleepy
+	name = "knockout mine"
+	desc = "An unarmed mine that releases N2O into the air when triggered. Nighty Night!"
+	mine_type = /obj/effect/mine/gas/n2o
+
+/obj/item/deployablemine/afterattack(atom/plantspot, mob/user, proximity)
+	if(!proximity)
+		return
+
+	if(!istype(plantspot,/turf/open)) // you can't plant a mine inside a wall or on a mob
+		return
+
+	if(isspaceturf(plantspot))
+		to_chat(user, "<span class='warning'>you cannot plant a mine in space!</span>")
+		return
+
+	if((istype(plantspot,/turf/open/lava)) || (istype(plantspot,/turf/open/chasm)))
+		to_chat(user, "<span class='warning'>You can't plant the mine here!</span>")
+		return
+
+	to_chat(user, "<span class='notice'>You start arming the [src]...</span>")
+	if(do_after(user, arming_time, target = src))
+		new mine_type(plantspot)
+		to_chat(user, "<span class='notice'>You plant and arm the [src].</span>")
+		log_combat(user, src, "planted and armed")
+		qdel(src)
```

Comment:
```suggestion
	if(!do_after(user, arming_time, target = src))
		return
	new mine_type(plantspot)
	to_chat(user, "<span class='notice'>You plant and arm the [src].</span>")
	log_combat(user, src, "planted and armed")
	qdel(src)
```
Make use of early returns.
You might want to check earlier if there's a `mine_type` defined. After all one could conceive non-functional mines existing.

Reply:
mine types should allready be defiend yep